### PR TITLE
Add support for SSL database connections

### PR DESCRIPTION
**Pull request by juan131**
Signed-off-by: juan131 <juan@bitnami.com>

## Description of the change

This PR adds the [Secure DB Connection](https://wordpress.org/plugins/secure-db-connection/) plugin to the WP example for **Stacksmith** and configure the corresponding SSL certificates to ensure that connections with the database are encrypted.

## Benefits

- Improves the security in the WP example.
- It allows using this example app on cloud providers that enables Database SSL Enforcement.

### CODE REVIEW

File: stacksmith/user-scripts/build.sh [hunk 1]
```diff
@@ -21,6 +21,12 @@ downloadWordPress() {
     wp --path=$installdir core download
 }
 
+downloadSSLCertificates() {
+    echo "==> Downloading SSL certificates..."
+    # ref: https://docs.microsoft.com/en-us/azure/mysql/howto-configure-ssl
+    curl -Lo '/opt/azure-db.crt.pem' https://www.digicert.com/CACerts/BaltimoreCyberTrustRoot.crt.pem
```
Comment (wojciechka) [inline stacksmith/user-scripts/build.sh]: Have you tried getting the CA certificates from `/etc/pki/ca-trust/extracted/openssl/ca-bundle.trust.crt`?

The file comes from `ca-certificates` RPM package that I believe is installed in all CentOS instances.

File: stacksmith/user-scripts/boot.sh [hunk 2]
```diff
@@ -2,8 +2,22 @@
 
 set -euo pipefail
 
+getSSLCA() {
+    if [ -f "/opt/stacksmith/stacksmith-scripts/extra/rds-combined-ca-bundle.pem" ]; then
+        echo "/opt/stacksmith/stacksmith-scripts/extra/rds-combined-ca-bundle.pem"
+    elif [ -f "/opt/azure-db.crt.pem" ]; then
```
Comment (wojciechka) [inline stacksmith/user-scripts/boot.sh]: If my comment about using a different CA certificate file is valid, what you could do is remove the `elif` completely and always provide the generic trusted CA bundle file.

File: stacksmith/user-scripts/build.sh (hunk 1 above)
Comment (juan131) [reply stacksmith/user-scripts/build.sh]: Microsoft recommends using their certificate to connect to the Azure database:
https://docs.microsoft.com/en-us/azure/mysql/howto-configure-ssl

I can confirm those certificates are in the VM and the database connection seems to work when using it:

```
$ ls /etc/pki/ca-trust/extracted/openssl/ca-bundle.trust.crt
/etc/pki/ca-trust/extracted/openssl/ca-bundle.trust.crt
```

It even works just enabling the plugin and adding (without indicating any CA):

```
/** Force SSL for database connections */
define( 'MYSQL_CLIENT_FLAGS', MYSQLI_CLIENT_SSL );
```

The plugins says:

```
Depending on the MySQL server setup the SSL certs used may not be in the trusted store, if that’s the case mysqli_ssl_set() needs to be called to set custom keys and certs before connect. This Plugin adds a custom DB class that allows these settings to be configured via custom constants.
```

My question is: even if it works, why are we going to use different certificates to the ones that Microsoft suggest to use?

Comment (wojciechka) [reply stacksmith/user-scripts/build.sh]: Perhaps the safest thing to do is to leave as is and perhaps just add `elif [ -f /etc/pki/ca-trust/extracted/openssl/ca-bundle.trust.crt ]` and use the cert from there if it exists.

I'd also add comments on why we're downloading the SSL cert, linking to https://docs.microsoft.com/en-us/azure/mysql/howto-configure-ssl - so this is more clear in the future

In that case I'll approve, as my comment for another `elif` is just a suggestion if it works :)

Comment (juan131) [reply stacksmith/user-scripts/build.sh]: > I'd also add comments on why we're downloading the SSL cert, linking to https://docs.microsoft.com/en-us/azure/mysql/howto-configure-ssl - so this is more clear in the future

The reference is already there. I can add a comment if it's necessary.

File: stacksmith/user-scripts/boot.sh (hunk 2 above)
Comment (juan131) [reply stacksmith/user-scripts/boot.sh]: Added as fallback

**Pull request merged**